contentcuration/contentcuration/views/admin.py

```diff
@@ -141,6 +142,149 @@ def get_all_users(request):
+    # return "hey!"
+    paginator = ChannelUserListPagination()
+    user_list = User.objects.prefetch_related('editable_channels').prefetch_related('view_only_channels').distinct()
+    if 'channel_id' in params:
+        channel = Channel.objects.get(pk=params['channel_id'])
+        user_list = user_list.filter(editable_channels__in=[channel])
+    
+    if AND_('is_chef') in params:
+        # import ipdb;ipdb.set_trace()
+        user_list = user_list.annotate(chef_channels=Count('editable_channels__ricecooker_version'))
+        if params.get(AND_('is_chef')) == 'True':
+            user_list = user_list.filter(chef_channels__gt=0)
+        else:
+            user_list = user_list.filter(chef_channels=0)
+
+    query = None
+    for prop in ['is_admin', 'is_active']:
+        if AND_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[AND_(prop)] == 'True' else ~q
+            query = q & query if query else q
+        if OR_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[OR_(prop)] == 'True' else ~q
+            query = q | query if query else q
+    
+    if query:
+        user_list = user_list.filter(query)
+
+    if params.get('sort_by') in ['email','first_name','last_name','channels']:
+        if params.get('ordering') == 'desc':
+            ordering = F(params.get('sort_by')).desc()
+        else:
+            ordering = F(params.get('sort_by')).asc()
+
+        user_list = user_list\
+                    .annotate(channels=Count('editable_channels'))\
+                    .order_by(ordering)
+
+    user_list_page = paginator.paginate_queryset(user_list, request)
+    user_serializer = AdminUserListSerializer(user_list_page, many=True)
+
+    return Response(user_serializer.data)
+
+channel_filters = {
+    'all': Q(),
+    'is_live': ~Q(deleted=True),
+    'is_published': Q(published=True),
+    'is_public': Q(public=True),
+    'is_private': ~Q(public=True),
+    'can_edit': Q(can_edit=True),
+    'is_staged': ~Q(staging_tree=None),
+    'is_ricecooker': ~Q(ricecooker_version=None),
+    'is_deleted': Q(deleted=True)
+}
+
+channel_sortable_keys = [
+    'name',
+    'id',
+    'priority', 
+    'users', 
+    # 'items',      # not working yet!
+    # 'modified',       # not working yet!
+    'created'
+]
+@login_required
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+def get_channels(request):
+    if not request.user.is_admin:
+        raise SuspiciousOperation("You are not authorized to access this endpoint")
+
+    params = request.GET
+    paginator = ChannelUserListPagination()
+    channel_list = Channel.objects.select_related('main_tree').prefetch_related('editors', 'viewers').distinct()    
+
+    if params.get('user_id'):
+        user = User.objects.get(pk=params['user_id'])
+        channel_list = channel_list.filter(editors__in=[user])
```

Comment:
Will this preserve the current "admins are implicit edit rights on everything" logic?

Currently on `/channels/administation/`, if you're admin you can see all channels (even ones you're not explicitly editor on), so it would be good to preserve that if the `get_channels` ends up used for that purpose.